### PR TITLE
add Equal function

### PR DESCRIPTION
**Pull request by Thor-wl**
ref: https://github.com/volcano-sh/volcano/issues/1525
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: pkg/scheduler/api/resource_info.go [hunk 1]
```diff
@@ -368,6 +368,32 @@ func (r *Resource) LessInSomeDimension(rr *Resource) bool {
...
+
+	if !equalFunc(leftResource.MilliCPU, rightResource.MilliCPU, minResource) || !equalFunc(leftResource.Memory, rightResource.Memory, minResource) {
+		return false
+	}
+
+	r.setDefaultValue(leftResource, rightResource, defaultValue)
+
+	for resourceName, leftValue := range leftResource.ScalarResources {
+		rightValue, _ := rightResource.ScalarResources[resourceName]
+		if leftValue != rightValue {
```
Comment (wpeng102) [inline pkg/scheduler/api/resource_info.go:390]: suggest also use `equalFunc` to compare

Comment (Thor-wl) [reply pkg/scheduler/api/resource_info.go:390]: > suggest also use `equalFunc` to compare

Good idea. THX.

Comment (Thor-wl) [reply pkg/scheduler/api/resource_info.go]: done

**Review by wpeng102 — APPROVED** [review]
/lgtm

**Pull request merged**